a8cteam51/plugin-autoupdate-filter
# Feature/killswitch

## Synopsis

Simplified version of the centralized settings here: https://github.com/a8cteam51/plugin-autoupdate-filter/pull/24

The point of this is to get an MVP up and running, with the most important feature, and then we can discuss and add the "Swiss Army Knife" features later.

## To Test

- This plugin is activated on opsoasis: https://github.com/a8cteam51/team51-autoupdate-settings
- The settings page is here, with a toggle switch: https://opsoasis.wpspecialprojects.com/wp-admin/tools.php?page=team51_autoupdate_settings_options
- The API endpoint is here: https://opsoasis.wpspecialprojects.com/wp-json/custom/v1/get_autoupdate_settings/

When that toggle switch is toggled on, then all autoupdates should stop happening, and a notice should appear on the top of the plugins page.

## Details

- **Caching:** The settings are cached, causing the killswitch to not immediately take effect. This is likely because the settings are retrieved when the object is created. However, I also tried it by retrieving the settings every single time the update filters ran, and it still seemed to not take effect immediately. I can deactivate/reactivate the Plugin Autoupdate Filter plugin to get it to register changes to the settings. Any thoughts on how to do this better are encouraged.
- **Naming of the killswitch:** as is being discussed in the other PR, I'm still waffling over the naming of the killswitch being confusing with the boolean return for the autoupdate filter itself. (i.e. Killswitch is "on" and return is "false")

Any other feedback is helpful, with the goal of getting out a minimum viable product that includes the killswitch, and we can add more features later :)
